Bottom: 34d92b470005510eca20804c79297e53d7da68d7
Top:    0468880bee12c5ffa22db468ab243daf13a571ee
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-12 01:31:08 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index eadf6fc..1a08467 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1154,7 +1154,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
